### PR TITLE
Make JSON and YAML output consistent across `list`, `components`, and `configurations` commands

### PR DESCRIPTION
**Pull request by shubham1172**
Signed-off-by: Shubham Sharma <shubhash@microsoft.com>

# Description

This PR fixes the output from `dapr list`, `dapr components`, and `dapr configurations`, to return a valid JSON or YAML all the time. This also adds UTs to cover the same.

## Behavior

No applications running:
```bash
$ dapr list --output json
[]
```

Single application running (**FIXED**):
```bash
$ dapr list --output json
[
  {
    "appId": "Test-App",
    "httpPort": 3500,
    "grpcPort": 50001,
    "appPort": 0,
    "metricsEnabled": true,
    "command": "",
    "age": "6s",
    "created": "2023-01-20 09:47.37",
    "daprdPid": 28708,
    "cliPid": 0,
    "maxRequestBodySize": 4,
    "httpReadBufferSize": 4
  }
]
```

Multiple applications running:
```bash
$ dapr list --output json
[
  {
    "appId": "Test-App-2",
    "httpPort": 3501,
    "grpcPort": 50002,
    "appPort": 0,
    "metricsEnabled": true,
    "command": "",
    "age": "4s",
    "created": "2023-01-20 09:48.36",
    "daprdPid": 29380,
    "cliPid": 0,
    "maxRequestBodySize": 4,
    "httpReadBufferSize": 4
  },
  {
    "appId": "Test-App",
    "httpPort": 3500,
    "grpcPort": 50001,
    "appPort": 0,
    "metricsEnabled": true,
    "command": "",
    "age": "1m",
    "created": "2023-01-20 09:47.37",
    "daprdPid": 28708,
    "cliPid": 0,
    "maxRequestBodySize": 4,
    "httpReadBufferSize": 4
  }
]
```

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #1171

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [X] Code compiles correctly
* [X] Created/updated tests
* [ ] Extended the documentation

### CODE REVIEW

File: utils/utils.go [hunk 1]
```diff
@@ -238,20 +237,16 @@ func MarshalAndWriteTable(writer io.Writer, in interface{}) error {
 }
 
 func PrintDetail(writer io.Writer, outputFormat string, list interface{}) error {
-	obj := list
```
Comment (philliphoff) [inline utils/utils.go:241]: As this is used for more than just `dapr list`, any concern about the impact of this change on other uses (e.g. component output)?

Comment (philliphoff) [reply utils/utils.go:241]: I was working on a [PR of my own](https://github.com/philliphoff/cli/tree/philliphoff-json-lists), mostly as practice in Go, but you beat me to it. :-)

I was concerned about backward compatibility, so my change selectively turned off list unwrapping only for JSON output of `dapr list`.

Comment (shubham1172) [reply utils/utils.go:241]: It's used by `dapr components -k` and `dapr configurations -k` to list components & configuration using JSON & YAML outputs, and these commands also had the same inconsistency issue. This change fixes them as well.

Comment (philliphoff) [reply utils/utils.go:241]: You'd know better than me. :-)

**Review by philliphoff — APPROVED** [review]
Looks good to me! Thumbs up for the new tests, too.

Might be worth ensuring this change is highlighted in the release notes, in case users are depending on the earlier "broken" behavior.

File: utils/utils.go (hunk 1 above)
Comment (shubham1172) [reply utils/utils.go:241]: > I was working on a [PR of my own](https://github.com/philliphoff/cli/tree/philliphoff-json-lists), mostly as practice in Go, but you beat me to it. :-)

<details>
  <summary>Haha, apologies!</summary>
  I was motivated to create this PR, hint: look at the PR number and my username :P
</details>

I think it should be okay to start being consistent in all the three outputs (`list`, `components`, `configurations`) and change the behavior. As you mentioned, we can explicitly call it out in the release notes. 

/fyi @mukundansundar

Comment (mukundansundar) [reply utils/utils.go:241]: The only question that I have here is for yaml should we print in the old format itself without the `-` that I see being added in the tests

Comment (shubham1172) [reply utils/utils.go:241]: Isn't this the same analogy as JSON? i.e. printing an object vs printing a list with the single object?

**Review by mukundansundar — APPROVED** [review]
lgtm

**Pull request merged**